src/main/tut/categories-great-and-small.md

```diff
@@ -0,0 +1,26 @@
+```Haskell
+class Monoid m where
+    mempty  :: m
+    mappend :: m -> m -> m
+```
+```scala
+trait Monoid[M] {
+  def mempty: M
+  def mappend(m1: M)(m2: M): M
```

Comment:
We need to decide how to best translate this (function vs method). In case of methods, I'm not sure curried parameter lists are the most idiomatic way. In any case, I've opened #5 to track this.

Reply:
Yeah, you're right - I thought it'd maybe be a more 'direct' translation to have multiple parameter lists, since you can do: 
```Haskell
f :: Int -> Int -> Int
f x y = x + y

g = uncurry f
:t g
g :: (Int, Int) -> Int
```
but `def mappend(m1: M)(m2: M): M` isn't really idiomatic + it would get ridiculous pretty quick with longer function signatures.